### PR TITLE
Test identity encoding for range requests

### PR DESCRIPTION
**Pull request by jakearchibald**
Tests for https://github.com/whatwg/fetch/pull/751.

### CODE REVIEW

File: fetch/range/general.window.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// META: script=resources/utils.js
+// META: script=/common/utils.js
+
+const onload = new Promise(r => window.addEventListener('load', r));
+
+// It's weird that browsers do this, but it should continue to work.
+promise_test(async t => {
+  await loadScript('resources/partial-script.py?pretend-offset=90000');
+  assert_true(self.scriptExecuted);
+}, `Script executed from partial response`);
```
Comment (jakearchibald) [inline fetch/range/general.window.js]: This test already existed, but I renamed the file and git didn't realise.

File: fetch/range/resources/range-sw.js [hunk 2]
```diff
@@ -108,7 +111,7 @@ function rangeHeaderPassthroughTest(event) {
     promise_test(async () => {
       await fetch(event.request);
       const response = await fetch('stash-take.py?key=' + key);
-      assert_equals(await response.json(), '"range-header-received"');
+      assert_equals(await response.json(), 'range-header-received');
```
Comment (jakearchibald) [inline fetch/range/resources/range-sw.js]: Bug-fixing an existing test.

File: fetch/range/sw.https.window.js [hunk 3]
```diff
@@ -4,28 +4,21 @@
...
-const SCOPE = 'resources/basic.html' + Math.random();
-
-function appendAudio(document, url) {
-  const audio = document.createElement('audio');
-  audio.muted = true;
-  audio.src = url;
-  audio.preload = true;
-  document.body.appendChild(audio);
-}
+const BASE_SCOPE = 'resources/basic.html?';
```
Comment (jakearchibald) [inline fetch/range/sw.https.window.js]: I'm now generating a unique scope for each test, it avoid the old service worker coming back from the dead in Safari.

File: fetch/range/resources/long-wav.py [hunk 4]
```diff
@@ -50,11 +50,28 @@ def main(request, response):
...
         # This is later collected using stash-take.py
-        request.stash.put(range_received_key, 'range-header-received', '/fetch/range/')
+        request.server.stash.put(range_received_key, 'range-header-received', '/fetch/range/')
+
+    if accept_encoding_key:
+        # Remove any current value
+        request.server.stash.take(
+            accept_encoding_key,
+            '/fetch/range/'
+        )
```
Comment (annevk) [inline fetch/range/resources/long-wav.py]: Why is this needed?

File: fetch/range/resources/long-wav.py [hunk 5]
```diff
@@ -50,11 +50,28 @@ def main(request, response):
     response.headers.set("Cache-Control", "no-cache")
 
     range_header = request.headers.get('Range', '')
+    range_header_match = range_header and re.search(r'^bytes=(\d*)-(\d*)$', range_header)
```
Comment (annevk) [inline fetch/range/resources/long-wav.py]: So this is either None (or False?) or a regular expression result?

**Review by annevk — COMMENTED** [review]
I don't see anything that I know to be wrong, but I have a couple minor questions.

File: fetch/range/resources/long-wav.py (hunk 4 above)
Comment (jakearchibald) [reply fetch/range/resources/long-wav.py]: Safari appears to make two requests in quick succession. Without this, an error is thrown, since it tries to put something in the stash when that key already exists.

File: fetch/range/resources/long-wav.py (hunk 5 above)
Comment (jakearchibald) [reply fetch/range/resources/long-wav.py]: Yep

**Review by annevk — APPROVED** [review]
